tools/dep_tree_resolver/dep_why.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python3
+
+# This tool uses the output of `go_deps.go` and tracks down the dependency
+# tree branches that import the specified module. This is useful when
+# tracking down what parent depenedency is causing a particular module
+# to be downloaded/used.
+
+import argparse
+
+
+def parse_line(line):
+    """
+    This method goes line-by-line in the file and returns the machine
+    usable fields from it.
+    """
+    trimmed_line = line.strip()
+    leading_spaces = len(line) - len(trimmed_line)
+
+    # Strip the minus
+    module_dep = trimmed_line[2:]
+
+    return int(leading_spaces / 4) + 1, module_dep
+
+
+def dep_why(dep_tree_filename, target_module):
+    """
+    This method recursively goes down the dependency tree and prints out
+    the branches that lead to the specified target module
+    """
+
+    print("Searching for '{}' in {}...".format(target_module, dep_tree_filename))
+
+    found = False
+    current_path = []
+    with open(dep_tree_filename) as infile:
+        for line in infile:
+            level, module_dep = parse_line(line)
+
+            if level <= len(current_path):
+                for _ in range(len(current_path) - level + 1):
+                    if len(current_path) > 0:
+                        current_path.pop()
+
```

Comment:
I think this is equivalent, right?
```suggestion
                current_path = current_path[:level+1]
```

I find it easier to follow when written that way (but maybe that is just me)

Reply:
Yeah that's much simpler and easier to follow! 

Fixed via https://github.com/DataDog/datadog-agent/pull/9308/commits/0985b2d57a3c059c25589cdf0fc0b56961eac6fd